actionpack/lib/action_dispatch/http/response.rb

```diff
@@ -32,27 +32,14 @@ module ActionDispatch # :nodoc:
   #    end
   #  end
   class Response
-    class Header < DelegateClass(Hash) # :nodoc:
-      def initialize(response, header)
-        @response = response
-        super(header)
-      end
-
-      def []=(k, v)
-        if @response.sending? || @response.sent?
-          raise ActionDispatch::IllegalStateError, "header already sent"
-        end
```

Comment:
Are we still protecting against setting headers after the response has started sending?

Reply:
To the extent that it's reasonable and possible, yes.

Reply:
By the way, there were plenty of ways to bypass `[]=` in the previous implementation... so I'm a little concerned the previous implementation was insufficient. e.g. `Hash#update`...